runtime/compiler/rpc/J9Server.hpp

```diff
@@ -54,9 +54,24 @@ class J9ServerStream : J9Stream
    template <typename ...T>
    std::tuple<T...> read()
       {
+      uint32_t version = 0;
       readBlocking(_cMsg);
       if (!_cMsg.status())
          throw StreamCancel();
+      //Version data will be present only in the first message for a connection.
+      if ((_versionCheckStatus == NOTDONE) &&
+            (version = _cMsg.version()))
+         {
+         if (!checkVersion(version))
+            {
+            _versionCheckStatus = FAILED;
+            throw StreamCancel();
```

Comment:
The message for StreamCancel is "compilation canceled by client". I don't think this is the best type of exception to be thrown here.